src/App.jsx

```diff
@@ -145,102 +69,68 @@ function App() {
         }
       }), {});
 
-      setColumns(updatedColumns);
+      await updateColumnsOrder(updatedColumns);
       return;
     }
 
-    // Se for arrasto de tarefas
-    const sourceColumn = columns[source.droppableId];
-    const destColumn = columns[destination.droppableId];
-    const sourceTasks = [...sourceColumn.tasks];
-    const destTasks = source.droppableId === destination.droppableId 
-      ? sourceTasks 
-      : [...destColumn.tasks];
-
-    const [removed] = sourceTasks.splice(source.index, 1);
-    destTasks.splice(destination.index, 0, removed);
-
-    setColumns({
-      ...columns,
-      [source.droppableId]: {
-        ...sourceColumn,
-        tasks: sourceTasks
-      },
-      [destination.droppableId]: {
-        ...destColumn,
-        tasks: destTasks
-      }
-    });
+    await moveTask(
+      source.droppableId,
+      destination.droppableId,
+      result.draggableId,
+      source.index,
+      destination.index
+    );
   };
 
-  const handleEmojiClick = (columnId, emojiObject) => {
-    setColumns({
-      ...columns,
-      [columnId]: {
-        ...columns[columnId],
-        emoji: emojiObject.emoji
-      }
+  const handleEmojiClick = async (columnId, emojiObject) => {
+    if (!user) return;
+    await updateColumn(columnId, {
+      emoji: emojiObject.emoji
     });
     setShowEmojiPicker(null);
   };
 
-  const handleColorChange = (columnId, color) => {
-    setColumns({
-      ...columns,
-      [columnId]: {
-        ...columns[columnId],
-        color: color
-      }
+  const handleColorChange = async (columnId, color) => {
+    if (!user) return;
+    await updateColumn(columnId, {
+      color: color
     });
     setShowColorPicker(null);
   };
 
+  const handleTaskClick = (columnId, task) => {
+    setSelectedTask({ ...task, columnId });
+  };
+
+  const handleTaskUpdate = async (taskId, updates) => {
+    if (!selectedTask?.columnId) return;
+    await updateTask(selectedTask.columnId, taskId, updates);
+  };
+
+  const handleTaskDelete = async (taskId) => {
+    if (!selectedTask?.columnId) return;
+    await deleteTask(selectedTask.columnId, taskId);
+  };
```

Comment:
![kody code-review](https://img.shields.io/badge/kody-code--review-brightgreen)

```suggestion
const handleTaskDelete = async (taskId) => {
  if (!selectedTask?.columnId) return;
  if (window.confirm('Are you sure you want to delete this task?')) {
    await deleteTask(selectedTask.columnId, taskId);
  }
};
```
Consider adding user confirmation before deleting a task to prevent accidental deletions.



<sub>Talk to Kody by mentioning @kody</sub>

<sub>Was this suggestion helpful? React with 👍 or 👎 to help Kody learn from this interaction.</sub><!-- kody-codereview -->&#8203;
&#8203;